compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -2034,8 +2034,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         };
 
         let sugg = match self.tcx.hir_maybe_get_struct_pattern_shorthand_field(expr) {
+            Some(_) if expr.span.from_expansion() => return false,
```

Comment:
I think we can remove more invalid suggestions if we add:

```rust
     if expr.span.from_expansion() {
            return false;
        }
```

at the start of `suggest_missing_unwrap_expect`.

Reply:
never mind, seems the above suggestion of https://github.com/rust-lang/rust/pull/141070/files#diff-a188fa3dba59dd225d4a65e08961d2937f6418bc6126716e001303bf946d6ee3R2018 could be valid for macro, such as https://github.com/chenyukang/rust/blob/c79bbfab78dcb0a72aa3b2bc35c00334b58bfe2e/tests/ui/typeck/question-mark-operator-suggestion-span.rs#L18